aws-transfer-agreement/aws-transfer-agreement.json

```diff
@@ -114,8 +114,12 @@
     "/properties/AgreementId",
     "/properties/Arn"
   ],
+  "createOnlyProperties": [
+    "/properties/ServerId"
+  ],
   "primaryIdentifier": [
-    "/properties/AgreementId"
+    "/properties/AgreementId",
+    "/properties/ServerId"
   ],
   "handlers": {
     "create": {
```

Comment:
Unrelated to this PR, but we tried deploying the agreement resource as a private type and we weren't able to create agreement without the `iam:PassRole` permission for the create handler.

Reply:
This would also apply to update handler.

Reply:
Right, `iam:PassRole` is required during create and update to ensure that the caller has permissions to the roles they're using in the request. Updated the schema to reflect that.